modules/core/src/main/java/org/apache/synapse/endpoints/Endpoint.java

```diff
@@ -156,6 +156,6 @@ public interface Endpoint extends ManagedLifecycle, SynapseArtifact, Nameable {
      *
      * @return JSONObject
      */
-    public JSONObject getJsonRepresentation();
+    public JSONObject getJsonRepresentation(MessageContext messageContext);
```

Comment:
Since this is a public method in an java interface, changing this would break custom endpoints written by users. Shall we try to avoid this if possible? Maybe we can overload this method and introduce a new one?

Reply:
Updated. Message context is no longer passed and the previous method in the interface is maintained.